### PR TITLE
[receiver/mongodb] Use metrics builder for resource metrics

### PR DESCRIPTION
**Pull request by dmitryax**
This change refactors the receiver to rely more on generated code.

The only functional change:
Metrics scope name is changed from `otelcol/mongodb` to `otelcol/mongodbreceiver`

### CODE REVIEW

**Comment by dmitryax** [comment]
Another issue uncovered after this change: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9760

**Review by TylerHelmuth — APPROVED** [review]
LGTM

**Review by djaglowski — APPROVED** [review]

File: receiver/mongodbreceiver/metadata.yaml [hunk 1]
```diff
@@ -1,5 +1,10 @@
 name: mongodbreceiver
 
+resource_attributes:
+  database:
+    description: The name of a database.
+    type: string
+
 attributes:
   database:
```
Comment (codeboten) [inline receiver/mongodbreceiver/metadata.yaml:9]: Is it expected to have the attribute and resource_attribute be a duplicate of one another?

Comment (dmitryax) [reply receiver/mongodbreceiver/metadata.yaml:9]: Yes, this is current behavior. I submitted a separate issue to address that https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9760

**Pull request merged**